### PR TITLE
:bug: Fix Target Group's name exceeding 32 characters

### LINKED ISSUE #4947: Do not exceed 32 characters limit for Target Group name
/kind bug

**What steps did you take and what happened:**
Using the LB's name as part of the Target Group's name has the potential of causing failures since both have a limit of 32 characters. For example, using `rdossant-installer-04-wn9qs-int` as the LB name, will result in the following error
```
time="2024-04-22T22:54:32Z" level=debug msg="\tfailed to create target group for load balancer: ValidationError: Target group name 'rdossant-installer-04-wn9qs-int-6443' cannot be longer than '32' characters"
```

**What did you expect to happen:**
Either CAPA goes back to using a hardcoded prefix for TGs or the TG's name is transformed so it fits the limit.

**Anything else you would like to add:**
Target Group name change was introduced in https://github.com/kubernetes-sigs/cluster-api-provider-aws/pull/4849

**Environment:**

- Cluster-api-provider-aws version: main @ 810bbf4eafd8fd6584e0c653b24c5f07f223b8e4
- Kubernetes version: (use `kubectl version`): 
- OS (e.g. from `/etc/os-release`):

### PR DESCRIPTION
**Pull request by r4f4**
**What type of PR is this?**
/kind bug

**What this PR does / why we need it**: 
Using the LB's name as part of the Target Group's name has the potential of causing failures since both have a limit of 32 characters. For example, using rdossant-installer-04-wn9qs-int as the LB name, will result in the following error
```
time="2024-04-22T22:54:32Z" level=debug msg="\tfailed to create target group for load balancer: ValidationError: Target group name 'rdossant-installer-04-wn9qs-int-6443' cannot be longer than '32' characters"
```

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #4947

**Special notes for your reviewer**:

**Checklist**:

- [ ] squashed commits
- [ ] includes documentation
- [X] includes [emojis](https://github.com/kubernetes-sigs/kubebuilder-release-tools?tab=readme-ov-file#kubebuilder-project-versioning)
- [ ] adds unit tests
- [ ] adds or updates e2e tests

**Release note**:

```release-note
Revert a change where the Target Group's name would use the Load Balancer's name as prefix, possibly causing it to exceed the 32 characters limit
```

### CODE REVIEW

**Comment by r4f4** [comment]
/cc @mtulio 
This is one possible solution. Feel free to suggest alternatives.

File: pkg/cloud/services/elb/loadbalancer.go [hunk 1]
```diff
@@ -276,7 +263,7 @@ func (s *Service) getAPIServerLBSpec(elbName string, lbSpec *infrav1.AWSLoadBala
 				Protocol: infrav1.ELBProtocolTCP,
 				Port:     infrav1.DefaultAPIServerPort,
 				TargetGroup: infrav1.TargetGroupSpec{
-					Name:        apiTargetGroupName,
+					Name:        fmt.Sprintf("apiserver-target-%d", time.Now().Unix()),
```
Comment (damdo) [inline pkg/cloud/services/elb/loadbalancer.go:266]: Ok this is currently 30 chars.
```
>>> len("additional-listener-1257894000")
30
```

**Review by damdo — COMMENTED** [review]
/lgtm

**Comment by mtulio** [comment]
> /cc @mtulio This is one possible solution. Feel free to suggest alternatives.

I believe it is fine reverting the subset of #4849 considering the main targets are attributes for the health check. For UX it would be interesting allowing or reusing some custom prefix (like cluster id), or truncating the name in the way that will not have duplicated values.

/lgtm

**Comment by damdo** [comment]
/assign @richardcase @vincepri @Ankitasw @dlipovetsky 
For approval

**Review by vincepri — APPROVED** [review]

**Pull request merged**